File: features/merchant/product_manage_list/src/main/java/com/tokopedia/product/manage/oldlist/view/activity/ProductManageActivity.java
package com.tokopedia.product.manage.oldlist.view.activity;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.play.core.splitcompat.SplitCompat;
import com.tokopedia.abstraction.base.view.activity.BaseActivity;
import com.tokopedia.abstraction.common.di.component.HasComponent;
import com.tokopedia.applink.ApplinkConst;
import com.tokopedia.applink.RouteManager;
import com.tokopedia.applink.internal.ApplinkConstInternalSellerapp;
import com.tokopedia.config.GlobalConfig;
import com.tokopedia.product.manage.oldlist.constant.AppScreen;
import com.tokopedia.product.manage.oldlist.view.fragment.ProductManageSellerFragment;
import com.tokopedia.user.session.UserSession;
import com.tokopedia.user.session.UserSessionInterface;

/**
 * Created by zulfikarrahman on 9/25/17.
 */
public class ProductManageActivity extends BaseActivity {

    public static final String TAG = ProductManageActivity.class.getSimpleName();

    public UserSessionInterface userSession;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        userSession = new UserSession(this);
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(newBase);
        SplitCompat.installActivity(this);
    }

    @Override
    public void onStart() {
        super.onStart();
        checkLogin();
    }

    @Override
    public void onBackPressed() {
        goToSellerAppDashboard();
        super.onBackPressed();
    }

    private void checkLogin() {
        if (!userSession.isLoggedIn()) {
            RouteManager.route(this, ApplinkConst.LOGIN);
            finish();
        } else if (!userSession.hasShop()) {
            RouteManager.route(this, ApplinkConst.HOME);
            finish();
        }
    }

    @Override
    public String getScreenName() {
        return AppScreen.SCREEN_MANAGE_PROD;
    }

    private void goToSellerAppDashboard() {
        if(GlobalConfig.isSellerApp()) {
            RouteManager.route(this, ApplinkConstInternalSellerapp.SELLER_HOME);
        }
    }
}

File: features/merchant/product_manage_list/src/main/java/com/tokopedia/product/manage/feature/list/view/activity/ProductManageActivity.kt
package com.tokopedia.product.manage.feature.list.view.activity

import android.content.Context
import android.os.Bundle
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.google.android.play.core.splitcompat.SplitCompat
import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity
import com.tokopedia.abstraction.common.di.component.HasComponent
import com.tokopedia.applink.RouteManager
import com.tokopedia.applink.internal.ApplinkConstInternalMarketplace
import com.tokopedia.applink.productmanage.DeepLinkMapperProductManage
import com.tokopedia.product.manage.common.util.ProductManageConfig
import com.tokopedia.product.manage.feature.list.di.ProductManageListComponent
import com.tokopedia.product.manage.feature.list.di.ProductManageListInstance
import com.tokopedia.product.manage.feature.list.view.fragment.ProductManageSellerFragment

open class ProductManageActivity : BaseSimpleActivity(), HasComponent<ProductManageListComponent> {

    companion object {
        private const val SCREEN_NAME = "Store - Manage product"
    }

    private val productManageSellerFragment by lazy {
        val uri = intent.data
        val filterId = uri?.getQueryParameter(DeepLinkMapperProductManage.QUERY_PARAM_FILTER).orEmpty()
        val searchKeyword = uri?.getQueryParameter(DeepLinkMapperProductManage.QUERY_PARAM_SEARCH).orEmpty()
        val tab = uri?.getQueryParameter(DeepLinkMapperProductManage.QUERY_PARAM_TAB).orEmpty()

        return@lazy when {
            filterId.isNotBlank() || searchKeyword.isNotBlank() || tab.isNotBlank() -> {
                if (filterId.isEmpty()){
                    ProductManageSellerFragment.newInstance(arrayListOf(), tab, searchKeyword)
                }else{
                    ProductManageSellerFragment.newInstance(arrayListOf(filterId), tab, searchKeyword)
                }
            }
            else -> {
                ProductManageSellerFragment()
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initInjector()

        if (!ProductManageConfig.IS_SELLER_APP) {
            window.decorView.setBackgroundColor(ContextCompat.getColor(this, com.tokopedia.unifyprinciples.R.color.Unify_Background))
        }
    }

    private fun initInjector() {
        component.inject(this)
    }

    override fun getNewFragment(): Fragment? {
        return productManageSellerFragment
    }

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(newBase)
        SplitCompat.installActivity(this)
    }

    override fun onBackPressed() {
        goToSellerAppDashboard()
        super.onBackPressed()
    }

    override fun getScreenName(): String {
        return SCREEN_NAME
    }

    override fun getComponent(): ProductManageListComponent {
        return ProductManageListInstance.getComponent(this)
    }

    private fun goToSellerAppDashboard() {
        if (ProductManageConfig.IS_SELLER_APP) {
            RouteManager.route(this, ApplinkConstInternalMarketplace.SELLER_APP_DASHBOARD)
        }
    }
}
